src/python/deepgnn/graph_engine/snark/converter/writers.py

```diff
@@ -93,13 +95,17 @@ def add(self, data: typing.Iterator[typing.Tuple[int, int, int, float, list]]):
                 self.node_type_count[typ] += 1
                 self.node_count += 1
             else:
+                if src != prev_node_id:
+                    self.node_writer.add(src, -1, [])
```

Comment:
how do we identify a correct node type between multiple partitions then?

Reply:
We do not want any of these added nodes to have valid types, just ghost nodes so they get added to the node map and neighbor index.

Reply:
I don’t see any changes to GetNodeType function. This pr will return ghost types along with the regular node types.

Reply:
https://github.com/microsoft/DeepGNN/blob/7c573e532642a680b7be0361443c8868366e7bb2/src/cc/lib/distributed/graph_engine.cc#L86

Reply:
Added support for remote and local